cylc/flow/scripts/cylc.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+# Copyright (C) 2008-2017 NIWA
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import os
+
+import click
+import pkg_resources
+
+from cylc.flow import __version__
+
+# These will be ported to python as click commands
+bash_commands = ["cylc-graph-diff", "cylc-jobscript", "cylc-scp-transfer"]
+
+# First step of the click port, the list won't be necessary after that
+command_list = bash_commands + list(
+    pkg_resources.get_entry_map("cylc-flow").get("console_scripts").keys()
+)
+
+category_list = [
+    "control",
+    "information",
+    "all",
+    "task",
+    "admin",
+    "preparation",
+    "discovery",
+    "utility",
+    "util",
+    "prep",
+    "con",
+    "info",
+]
+
+
+def execute_cmd(cmd, *args):
+    # Replace the current process with that of the sub-command.
+    try:
+        os.execvp(cmd, [cmd] + list(args))  # nosec
+    except OSError as exc:
+        if exc.filename is None:
+            exc.filename = cmd
+        raise SystemExit(exc)
+
+
+def print_version(ctx, param, value):
+    if not value or ctx.resilient_parsing:
+        return
+    click.echo(__version__)
+    ctx.exit()
+
+
+CONTEXT_SETTINGS = dict(
+    token_normalize_func=lambda x: x.lower(), ignore_unknown_options=True
+)
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.argument("command")
+@click.argument("cmd-args", nargs=-1)
+@click.option("--help", "-h", "help_", is_flag=True)
+@click.option(
+    "--version",
+    "-v",
```

Comment:
I think this is causing several tests to fail, e.g. `tests/validate/28-fail-max-active-cycle-points-zero.t`. The issue is that it runs `cylc validate -v $suite-name`. The `-v` in that case is for **verbose**.

Here's the output running with Cylc 8.0a1 (similar to Cylc 7.8.x but colored):

```
(venv) kinow@kinow-VirtualBox:~/Development/python/workspace/cylc-flow$ cylc validate five
WARNING - deprecated graph items were automatically upgraded in 'suite definition':
WARNING -  * (8.0.0) [scheduling][dependencies][X][graph] -> [scheduling][graph][X] - for X in:
	PT12H
	R1
Valid for cylc-8.0a1
(venv) kinow@kinow-VirtualBox:~/Development/python/workspace/cylc-flow$ cylc validate -v five
DEBUG - Loading site/user config files
DEBUG - Reading file /home/kinow/cylc-suites/five/suite.rc
WARNING - deprecated graph items were automatically upgraded in 'suite definition':
WARNING -  * (8.0.0) [scheduling][dependencies][X][graph] -> [scheduling][graph][X] - for X in:
	PT12H
	R1
DEBUG - Expanding [runtime] namespace lists and parameters
DEBUG - Parsing the runtime namespace hierarchy
DEBUG - Parsing [special tasks]
DEBUG - Parsing the dependency graph
DEBUG - Configuring internal queues
WARNING - naked dummy tasks detected (no entry under [runtime]):
	+       prep
	+       foo
	+       bar
DEBUG - Checking for defined tasks not used in the graph
DEBUG - Checking [visualization] node attributes
Instantiating tasks to check trigger expressions
  + prep.20130808T0000Z ok
  + foo.20130808T0000Z ok
  + bar.20130808T0000Z ok
Valid for cylc-8.0a1
```

And here's where it is defined: https://github.com/cylc/cylc-flow/blob/f902059452fecde5a3c2928be30fcfc34f20ab38/cylc/flow/option_parsers.py#L148

Running `cylc validate -v five` (five is my suite/workflow) with this branch, I get the Cylc version, which is the output expected of `cylc -v`.

@sgaist I think we need to find a way to use the `cylc-$subcommand` parameters instead here?

Reply:
Well spotted!

Reply:
My bad ! I misread the version vs verbose handling.